### DISCUSSION TITLE
Comparison between systemd-nspawn and LXC

### DISCUSSION THREAD

**Discussion by foxtrotcz** [Jip-Hop/jailmaker#214] [General]
Hi,
I just want to say I am very much looking forward when you publish the comparison of systemd-nspawn and LXC as written here:
https://github.com/Jip-Hop/jailmaker?tab=readme-ov-file#comparison

When I tried to search for differences it wasnt hard to understand difference between VM, Docker and LXC. But I just cannot find any in-depth explanation about differences between LXC and systemd-nspawn.
I could just find that systemd-nspawn is simpler and has less features than LXC but otherwise they should be similar.
And maybe LXC has bigger community around it? Proxmox uses LXC and it also seems very active on their GitHub.

So if you can write some explanation about difference in features and advantages/disadvantages, that would be awesome.
Thanks.

**Comment by Jip-Hop** [comment]
I don't have enough experience with LXC to write an in depth comparison. This may very well remain a TODO forever >.< The biggest downside of nspawn is that the container guest needs to run systemd too to properly integrate. LXC doesn't have this downside. Regarding LXC I wanted to add there are multiple 'driver' implementations, so ups/downs also depend on which one you use. E.g. https://libvirt.org/drvlxc.html

> The libvirt LXC driver has no dependency on the LXC userspace tools hosted on sourceforge.net.

I think proxmox has its own (semi) custom implementation of LXC too.
